packages/components/src/internal/components/domainproperties/dataclasses/DataClassDesigner.tsx

```diff
@@ -85,11 +85,14 @@ class DataClassDesignerImpl extends PureComponent<Props & InjectedBaseDomainDesi
         if (this.state.model.isNew && isSampleManagerEnabled()) {
             const response = await this.props.loadNameExpressionOptions(this.state.model.containerPath);
 
-            this.setState(
-                produce((draft: Draft<State>) => {
-                    draft.model.nameExpression = response.prefix;
-                })
-            );
+            if (response.prefix) {
+                this.setState(
+                    produce((draft: Draft<State>) => {
+                        draft.model.nameExpression = response.prefix + (draft.model.nameExpression ? draft.model.nameExpression : '');
```

Comment:
The code here is within an if block that specifies `this.state.model.isNew`. Do we have a case in which there exists a non-null `draft.model.nameExpression` for a new data class?

Reply:
Yes, for LKSM sources (data classes), the default nameExpression is 'Source-${genId}'